cards/src/main/java/com/decathlon/vitamin/compose/cards/VitaminCards.kt

```diff
@@ -0,0 +1,258 @@
+        shape: Shape = RoundedCornerShape(4.dp),
+    ) = Card(modifier = modifier, shape = shape) {
+        Column {
+            if (actionContent != null) {
+                SideImageActionContent.actionContent()
+            }
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.spacedBy(16.dp),
+                modifier = Modifier
+                    .padding(
+                        top = if (actionContent != null) {
+                            0.dp
+                        } else {
+                            16.dp
+                        },
+                        bottom = 16.dp,
+                        start = 16.dp,
+                        end = 16.dp
+                    )
+            ) {
+                CardImage(
+                    painter = painter,
+                    contentDescription = contentDescription,
+                    isSideImage = true,
+                    modifier = Modifier.weight(weight = 1f).aspectRatio(ratio = 1f)
+                )
+                Box(Modifier.weight(weight = 3f)) {
+                    SideImageContent.content()
+                }
+            }
+        }
+    }
+
+    /**
+     * A card without image
+     * @param content The content of the card. Can be a composable from [WithoutImageContent] or a custom composable.
+     * @param modifier The `Modifier` to be applied to the all card.
+     * @param shape to change the shape of the card.
+     */
+    @Composable
+    fun WithoutImage(
+        content: (@Composable WithoutImageContent.() -> Unit),
+        modifier: Modifier = Modifier,
+        shape: Shape = RoundedCornerShape(4.dp),
+    ) = Card(modifier = modifier, shape = shape) {
+        Box(modifier = Modifier.padding(16.dp)) {
+            WithoutImageContent.content()
+        }
+    }
+}
+
+@Composable
+internal fun CardImage(
+    painter: Painter,
+    contentDescription: String?,
+    modifier: Modifier = Modifier,
+    overlay: @Composable (() -> Unit)? = null,
+    imageOverlayAlignment: Alignment = Alignment.TopStart,
+    withGradient: Boolean = false,
+    isSideImage: Boolean = false,
+) {
+    Box(modifier = modifier) {
+        Image(
+            painter = painter,
+            contentDescription = contentDescription,
+            contentScale = ContentScale.FillWidth,
+            modifier = when {
+                withGradient -> {
+                    Modifier
+                        .fillMaxWidth().withGradient(vtmnLightColorPalette.vtmnBackgroundPrimaryReversed)
+                }
+                isSideImage -> {
+                    Modifier.fillMaxSize()
+                }
+                else -> {
+                    Modifier.fillMaxWidth()
+                }
+            }
+        )
+        overlay?.let {
+            Box(
+                Modifier
+                    .padding(8.dp)
+                    .align(imageOverlayAlignment)
```

Comment:
Why there is a padding and a alignment here? The overlay shouldn't recover all the surface of the image?

Reply:
imageOverlayAlignment is used if you want to add an overlay on the image, for example some tags. I gave in the implementation the opportunity to align the overlay

<img width="575" alt="showing a card with a tag on overlay at top left" src="https://user-images.githubusercontent.com/3006608/173329923-8bf378d1-9cd6-4b06-ad40-311247d93710.png">